handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -1936,22 +1931,24 @@ public void handlerAdded(final ChannelHandlerContext ctx) throws Exception {
             // If we weren't able to include client_hello in the TCP SYN (e.g. no token, disabled at the OS) we have to
             // flush pending data in the outbound buffer later in channelActive().
             final ChannelOutboundBuffer outboundBuffer;
-            needsFlush |= fastOpen && ((outboundBuffer = channel.unsafe().outboundBuffer()) == null ||
-                    outboundBuffer.totalPendingWriteBytes() > 0);
+            if (fastOpen && ((outboundBuffer = channel.unsafe().outboundBuffer()) == null ||
+                    outboundBuffer.totalPendingWriteBytes() > 0)) {
+                setState(STATE_NEEDS_FLUSH);
```

Comment:
(Tracking down some Netty-related issue led me to this PR, and led me to do a close reading.  I noticed this and didn't want it to get lost in case it is valuable. It is also possible I've misread something.)

On this line, the prior version has:
```
needsFlush |= ...
```
…so shouldn't there be a corresponding `clearState(STATE_NEEDS_FLUSH)` if the elaborate condition does _not_ hold?

Again this is just from a flyby code read so I may have missed something deeper.

Reply:
That state is cleared when the flush happens. The effect of `|=` is to assign to the LHS the bitwise-OR of LHS and RHS. This PR does not change that behaviour. It just consolidates many fields into a single bit-field.